spec/forms/pages/conditions_form_spec.rb

```diff
@@ -32,15 +32,24 @@
 
   describe "#submit" do
     context "when validation pass" do
-      it "creates a condition" do
+      before do
         ActiveResource::HttpMock.respond_to do |mock|
           mock.post "/api/v1/forms/1/pages/2/conditions", post_headers, { id: 2 }.to_json, 200
         end
 
         page.id = 2
         conditions_form.answer_value = "Rabbit"
         conditions_form.goto_page_id = 4
+      end
+
+      it "calls assign_skip_to_end" do
+        allow(conditions_form).to receive(:assign_skip_to_end)
+        conditions_form.submit
+
+        expect(conditions_form).to have_received(:assign_skip_to_end).exactly(1).times
+      end
```

Comment:
I think this is testing implementation details rather than behaviour. What is the expected behaviour we expecting to happen when submit is called with specific params passed in.

Reply:
Do I need to test that? I'm testing the method itself in the `describe "#assign_skip_to_end"` block later in the file, so if I test it again in this test I feel like I'll just be repeating those cases

Reply:
ah sorry, I didn't spot that when I was reviewing this code. 

Ok, I see you have united tested the `assign_skip_to_end`.